Corrfunc/utils.py

```diff
@@ -1029,24 +1029,33 @@ def prep(w,x):
 @contextmanager
 def sys_pipes():
     '''
-    We can use the Wurlitzer package to redirect stdout and stderr
-    from the command line into a Jupyter notebook.  But if we're not
-    in a notebook, this isn't safe because we can't redirect stdout
-    to itself.  This function is a thin wrapper that checks if the
-    stdout/err streams are TTYs and enables output redirection
-    based on that.
+    In a Jupyter notebook, Python's ``sys.stdout`` and ``sys.stderr`` are redirected
+    so output ends up in cells.  But C extensions don't know about that!  Wurlitzer
+    uses os.dup2 to redirect fds 1 & 2 to the new location and restore them on return,
+    but will cause the output to hang if they were not already redirected.  It seems
+    we can compare Python's ``sys.stdout`` to the saved ``sys.__stdout__`` to tell
+    if redirection occurred.  We can also check if the output is a TTY, but that's
```

Comment:
Not sure I understand the TTY comment - arent you checking `sys.stdout/stderr.isatty`?  

From reading the wurlitzer source, seems that the default stdout/stderr is set to `None` - which is what the if condition is setting. As in, i am not sure why there would be a hang bug when the normal fd (1 & 2) are duplicated to non-blocking (i.e., in the lines here https://github.com/minrk/wurlitzer/blob/main/wurlitzer.py#L198-L218) and restored at the end.

Reply:
I'm not totally clear on why there would be a hang either---seems like it's something about the combination of unbuffered output and self-redirection.  The latter, at least, is warned against in the Wurlitzer comments (https://github.com/minrk/wurlitzer/blob/main/wurlitzer.py#L432), which is why I added these checks.  The `if` condition passes the kwarg `None` to Wurlitzer, which just tells it not to do anything to that pipe.

Reply:
This previous comment was mostly noting a potential inconsistency between the comment and the actual code (Line 1051) that does check for `isatty`

Reply:
Oh, I see. Yes, I'll update the comment to reflect the code!